packages/itwin/tree-widget/README.md

```diff
@@ -200,6 +242,54 @@ Available header buttons:
 - `ModelsTreeComponent.HideAllButton` makes all categories hidden.
 - `ModelsTreeComponent.InvertButton` inverts display of all categories.
 
+#### Custom categories tree
+
+This package provides building blocks for custom categories tree:
+
+- `useCategoriesTree` - hook for creating and managing categories tree state.
+- `useCategoriesTreeButtonProps` - hook for creating props for categories tree buttons.
+
+
+Example:
+
+```tsx
+function CustomCategoiresTreeComponent({ imodel, viewport, getSchemaContext, selectionStorage }: CustomCategoriesTreeProps) {
+  const buttonProps = useCategoriesTreeButtonProps({ imodel, viewport });
+  const { categoriesTreeProps, rendererProps } = useCategoriesTree({ activeView: viewport });
+
+  return (
+    <TreeWithHeader buttons={[<CategoriesTreeComponent.ShowAllButton {...buttonProps} />, <CategoriesTreeComponent.HideAllButton {...buttonProps} />]}>
+      <VisibilityTree
+        {...categoriesTreeProps}
+        getSchemaContext={getSchemaContext}
+        selectionStorage={selectionStorage}
+        imodel={imodel}
+        treeRenderer={(props) => <CustomCategoriesTreeRenderer {...props} {...rendererProps} />}
+      />
+    </TreeWithHeader>
+  );
+}
+
+type VisibilityTreeRendererProps = ComponentPropsWithoutRef<typeof VisibilityTreeRenderer>;
+type CustomCategoriesTreeRendererProps = Parameters<ComponentPropsWithoutRef<typeof VisibilityTree>["treeRenderer"]>[0];
+
+function CustomModelsTreeRenderer(props: CustomCategoriesTreeRendererProps) {
```

Comment:
```suggestion
function CustomCategoriesTreeRenderer(props: CustomCategoriesTreeRendererProps) {
```

Reply:
Filed https://github.com/iTwin/viewer-components-react/issues/1001 to help us capture such issues